### PR TITLE
Added validation to request builder ensure the header model builds a valid request

### PR DESCRIPTION
**Pull request by VeryStrongFingers**

### CODE REVIEW

File: src/Webhooks/RequestBuilder.php [hunk 1]
```diff
@@ -83,4 +86,33 @@ public function build(WebhookRequestInterface $webhookRequest): RequestInterface
...
+        $invalidHeaders = [];
+
+        /**
+         * Whilst this check should not be required due to type-hinting, due to magic setters on the model not enforcing
+         * strict type-hinting, it is possible that this array may contain 'mixed' values
+         */
+        foreach ($webhookRequest->getRequestHeaders() as $header => $value) {
+            /** @var mixed $value */
+            if (\is_scalar($value) === false) {
+                $invalidHeaders[] = $header;
```
Comment (merk) [inline src/Webhooks/RequestBuilder.php]: Just throw in here?

Comment (MitchellMacpherson) [reply src/Webhooks/RequestBuilder.php]: @merk throw once per invalid header structure, instead of collecting them all?

Comment (sjdaws) [reply src/Webhooks/RequestBuilder.php]: Agreed

**Review by sjdaws — DISMISSED** [review]

File: src/Webhooks/RequestBuilder.php (hunk 1 above)
Comment (merk) [reply src/Webhooks/RequestBuilder.php]: You're not capturing them all, you're just throwing a generic exception with no actual detail of which headers failed.

Simpler to just throw as soon as you see a bad one?

**Pull request closed without merging**